builtin/logical/pkiext/pkiext_binary/acme_sh.go

```diff
@@ -0,0 +1,81 @@
+package pkiext_binary
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	"github.com/hashicorp/vault/sdk/helper/docker"
+)
+
+const ACMEshContainerfile = `
+FROM ubuntu:latest
+
+RUN apt update && DEBIAN_FRONTEND="noninteractive" apt install -y openssl socat curl coreutils dnsutils tzdata sed tar jq libidn2-0 openssh-client git cron
+
+RUN git clone https://github.com/acmesh-official/acme.sh.git /acme.sh
+WORKDIR /acme.sh
+
+RUN ./acme.sh --install --home /etc/acmesh --config-home /etc/ssl/data --cert-home /etc/ssl/certs --accountemail "webmaster@dadgarcorp.com"
+`
+
+var (
+	acmeShRunner             *docker.Runner
+	buildAcmeShContainerOnce sync.Once
+)
+
+func buildACMEshContainer(t *testing.T, network string) {
+	bCtx := docker.NewBuildContext()
+
+	imageName := "vault_pki_acme_acme_sh_integration"
+	imageTag := "latest"
+	containerName := "vault_acme_sh_container"
+
+	var err error
+	acmeShRunner, err = docker.NewServiceRunner(docker.RunOptions{
+		ImageRepo:     imageName,
+		ImageTag:      imageTag,
+		ContainerName: containerName,
+		NetworkName:   network,
+		// We want to run sleep in the background so we're not stuck waiting
+		// for the default ubuntu container's shell to prompt for input.
+		//
+		// We choose a slightly longer default sleep, 600s = 10min, as we
```

Comment:
Ten minutes is a long time to wait for a test,  is there any way to fail that sooner?

Reply:
This is for when the tests fail in a way that the automatic cleanup of the container doesn't work; rather than leaving an opening server (like e.g., we do with `nginx`), this ensures the container gets automatically terminated after a long while.

For regular operation, the `runner.Stop(context.Background(), result.Container.ID)` should immediately clean up the container.